protocol_driver_double_barrel.cc

```diff
@@ -13,38 +13,35 @@
 // limitations under the License.
 
 #include "protocol_driver_double_barrel.h"
-//#include "protocol_driver_allocator.h"
+
+#include "protocol_driver_allocator.h"
 #include "absl/base/internal/sysinfo.h"
 #include "distbench_utils.h"
 #include "glog/logging.h"
 #include "google/protobuf/repeated_field.h"
 
 namespace distbench {
 
-absl::StatusOr<std::unique_ptr<ProtocolDriver>> AllocateProtocolDriver(
-    ProtocolDriverOptions opts, int* port, int tree_depth = 0);
-
 // ProtocolDriver ===============================================
 ProtocolDriverDoubleBarrel::ProtocolDriverDoubleBarrel(int tree_depth) {
   tree_depth_ = tree_depth;
 }
 
+typedef google::protobuf::RepeatedPtrField<NamedSetting> ServerSetting;
+
 absl::Status ProtocolDriverDoubleBarrel::Initialize(
     const ProtocolDriverOptions& pd_opts, int* port) {
 
   auto pdo = pd_opts;
-  std::string subordinate_driver_type = "";
-  google::protobuf::internal::RepeatedPtrIterator
-      <distbench::NamedSetting> iter_subordinate;
-  for (auto iter=pdo.server_settings().begin();
-       iter!=pdo.server_settings().end();
-       iter++) {
-    if (iter->name()=="next_protocol_driver") {
-      subordinate_driver_type = iter->string_value();
-      break;
-    }
+  auto server_settings = pdo.mutable_server_settings();
+  ServerSetting::iterator next_protocol_driver_it;
+  for (auto it=server_settings->begin(); it!=server_settings->end(); it++) {
+      if (it->name()=="next_protocol_driver") {
+        next_protocol_driver_it = it;
```

Comment:
what happens if this is not found, you are probably erasing an invalid iterator on line 44 below.

Reply:
Fixed